### PR TITLE
Fixed subp import

### PR DESCRIPTION
**Pull request by sshedi**
Made minor code improvements in getting subp module

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>

### CODE REVIEW

**Comment by jfroche** [comment]
You are removing the test `if hasattr(subp, 'subp'):` which verifies if subp package has a subp function.
We are running `Ubuntu 18.04` with `cloud-init 20.2-45` and we have this:
```python
>>> from cloudinit import subp
>>> subp.subp
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'cloudinit.subp' has no attribute 'subp'
>>> from cloudinit import util
>>> util.subp
<function subp at 0x7fae4157a730>
```
So we have a subp package but no subp function, so we should be using `cloudinit.util`

**Comment by sshedi** [comment]
@jfroche thanks, I missed it. Will send an update soon. It's weird what's been done in cloud-init, not moving a module fully in one shot. Simple added to further complications.

**Comment by yvespp** [comment]
Maybe this would be cleaner/easier to read:
```
try:
    from cloudinit.subp import subp, ProcessExecutionError
except ImportError:
    from cloudinit.util import subp, ProcessExecutionError
```
and call the methods directly (remove all `subp_module` references)

**Comment by skyscooby** [comment]
Could we prioritise merging one of these fixes please? master is broken on latest ubuntu 18.04 cloud-init version at the moment.

**Comment by skyscooby** [comment]
It would really help the users of this module to make #2 happen.. remember everyone's OS moves forward as they yum/apt update their code.. this module just sits in the python directory unmaintained and gets stale.. either this project needs to start building deb/rpm and host a repository so fixes can be released... or we need to get this code into the cloud-init project proper so they can deal with these module changes directly before releasing. 

Anyone relying on this module at the moment who updates their cloud-init package is in for serious pain on their next reboot.

**Comment by akutz** [comment]
We are trying. The problem is fixing it so that it works across all the various distros (see my earlier note about setting up automated testing pipelines). 

I try very much to make this DS backwards compatible so that it will work with all versions of CI on all distros. It is just challenging to cover everything. 

I merged two PRs already, but each time new issues were identified issues on additional other distro/cloud-init combinations.

So this time I’m making sure we have are not merging a fix that just breaks something else.

**Comment by sshedi** [comment]
> Maybe this would be cleaner/easier to read:
> 
> ```
> try:
>     from cloudinit.subp import subp, ProcessExecutionError
> except ImportError:
>     from cloudinit.util import subp, ProcessExecutionError
> ```
> 
> and call the methods directly (remove all `subp_module` references)

👍 will update my PR, thanks for the input

**Comment by sshedi** [comment]
@yvespp @akutz @jfroche Please give it a try, let's finish this one.

File: DataSourceVMwareGuestInfo.py [hunk 1]
```diff
@@ -326,15 +321,15 @@ def get_guestinfo_value(key):
 
     if data_access_method == VMWARE_RPCTOOL:
         try:
-            (stdout, stderr) = subp_module.subp(
+            (stdout, stderr) = subp.subp(
```
Comment (yvespp) [inline DataSourceVMwareGuestInfo.py]: You have to use the `subp` method directly without the module, e.g `(stdout, stderr) = subp(`
Same for the other calls.

**Review by yvespp — CHANGES_REQUESTED** [review]

File: DataSourceVMwareGuestInfo.py (hunk 1 above)
Comment (sshedi) [reply DataSourceVMwareGuestInfo.py]: Good point. Fixed it.

**Review by yvespp — APPROVED** [review]
Looks got to me now, tested it with Ubuntu 20.04

**Comment by LinAnt** [comment]
Can confirm this works with Ubuntu 20.04 on vSphere 6.7u3

**Comment by angelbarrera92** [comment]
One more here to confirm it works with Ubuntu 18.04. Is there any plan to release it?

**Comment by nickvth** [comment]
@akutz and @yvespp can you review this change and merge if possible? 
We are waiting for this fix.

**Comment by yvespp** [comment]
> @akutz and @yvespp can you review this change and merge if possible?
> We are waiting for this fix.

I don't have merge permission.

**Comment by akutz** [comment]
> > @akutz and @yvespp can you review this change and merge if possible?
> > We are waiting for this fix.
> 
> I don't have merge permission.

I was waiting to ensure this was properly tested this time. Merged.

**Comment by nickvth** [comment]
> > > @akutz and @yvespp can you review this change and merge if possible?
> > > We are waiting for this fix.
> > 
> > 
> > I don't have merge permission.
> 
> I was waiting to ensure this was properly tested this time. Merged.

Thanks @akutz @yvespp @sshedi

**Comment by sshedi** [comment]
Thanks all for getting involved in this and all the valuable suggestions. Liked it. It was a good learning experience.

**Pull request merged**